### DISCUSSION TITLE
Public accessible app - navigation menu is missing

### DISCUSSION THREAD

**Discussion by hheedfeld** [Budibase/budibase#7922] [Help]
Hi,

I've created a public accessible app, which is working very well, but I'm not able to make the navigation menu visible for unauthenticated users. What am I missing?

Thank for any hints,
Henning

**Comment by rpmcoding** [comment]
Have you navigated to the particular screen in question, then set access = Public and ensured the show navigation toggle is set as on?

**Reply by hheedfeld** [reply]
Yes, access = Public is set and "show navigation" is on.
Public (= unauthorized) access to to screen itself works fine, but the navigation menu is not shown.

**Reply by rpmcoding** [reply]
Does the navigation show when you are a logged in user as opposed to an unauthenticated user?

**Reply by hheedfeld** [reply]
Yes, the navigation shows up when I'm logged in.

**Reply by rpmcoding** [reply]
Weird.  Are you on version 1.4.7?

**Reply by hheedfeld** [reply]
Strange version info:
"This app is currently using version 1.4.7, but version 1.4.3 is available. Updates can contain new features, performance improvements and bug fixes."

**Reply by rpmcoding** [reply]
You get that message when you click check for updates at builder/portal/settings/update ?

**Comment by hheedfeld** [comment]
no there is a message "Update available" in the header of the app designer (see screenshot)
![Bildschirmfoto 2022-09-23 um 12 46 28](https://user-images.githubusercontent.com/12691334/191944857-f0fe0eb7-305a-486c-b40d-7eee55833ba9.png)

**Reply by rpmcoding** [reply]
That's odd.  Perhaps try updating to 1.4.7 at /builder/portal/settings/update and also the manage > Settings page for the app itself.  That way hopefully both are aligned.

**Comment by hheedfeld** [comment]
/builder/portal/settings/update -> Current version 1.4.7

App version -> 1.4.7

Still the same issue

**Reply by rpmcoding** [reply]
Damn, thought that would fix it.  Sorry, I'm all out of ideas.  I'm sure someone else will chime in here though.

**Comment by hheedfeld** [comment]
Any other guys with ideas?

**Comment by pbarmak** [comment]
I just upgraded to v2.0.3 and am seeing the same thing. Public links in navigation do not show in the published app. The funny thing is they do show in the dev preview when I preview as public. I think there is a bug here. @hheedfeld, did you create an Issue for this yet?

**Reply by pbarmak** [reply]
I also have an app still on version 1.4.3 (while core is on 2.0.3) and that is also hiding the public links now (it wasn't an issue before I upgraded core to 2.0.3, from what I know).

**Reply by kmeesters** [reply]
Actually have the same problem, visiting individual pages is not a problem as public visitor (using incognito window). But the navigation bar is not populated. I also see in the console several 403 errors:
`Failed to load resource: the server responded with a status of 403 ()
/api/global/self:1          Failed to load resource: the server responded with a status of 403 ()
budibase-client.js?v=2.0.37:22 [Client] HTTP 403 on GET:https://bb.propus.dev/api/global/self
onError @ budibase-client.js?v=2.0.37:22
budibase-client.js?v=2.0.37:39 Error: <foreignObject> attribute height: A negative value is not valid. ("-23.96666666666666")
value @ budibase-client.js?v=2.0.37:39
budibase-client.js?v=2.0.37:39 Error: <svg> attribute height: A negative value is not valid. ("-23.96666666666666")
value @ budibase-client.js?v=2.0.37:39`

I checked the permissions of all screen and tables (using internal DB). And they are all set to public. It seems really not be able 
to access the data related to the navigation. 

Addendum: Version 2.0.37

**Comment by Joe1080** [comment]
This worked for me: Go to "Configure Links" in navigation and set the access to _'minimum role'_
Then it was visible for users using public access.

**Reply by Joe1080** [reply]
@hheedfeld try this and let me know.

**Reply by kmeesters** [reply]
Mind blown: that works! So change it from 'Public' to 'Minimum role' (yes, it is counter-intuitive), now it works!
Seems indeed like a bug?

**Reply by Joe1080** [reply]
Probably is, but I'll take it as long as it works fine.

**Reply by pbarmak** [reply]
There is an open issue for this: #8031

**Comment by hheedfeld** [comment]
works for me, too!
Weird.
Thanks!
